File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 1
    for i in range(1, n+1):
        sum = i
        for j in range(i+1, n+1):

            if sum == n:
                answer += 1
                break
            elif sum > n:
                break 
            sum += j                  
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
  let answer = 0;
  let now = 1, cnt = 1, plus = 0;

  while(cnt <= n) {
    plus = plus + now;
    if(plus >= n) {
      if(plus === n) answer++;
      plus = 0;
      cnt++;
      now = cnt;
    } else {
      now++;
    }
  }

  return answer;
}